### PR TITLE
Dev 1.3

### PR DESCRIPTION
**Pull request by RapidfireCRH**
Moving debug messages to one method in RseData

### CODE REVIEW

File: RseData.py [hunk 1]
```diff
@@ -387,3 +388,15 @@ def initialize(self):
             for _row in json.loads(response):
                 rseProject = RseProject(_row["id"], _row["action_text"], _row["project_name"], _row["explanation"], _row["enabled"])
                 self.projectsDict[rseProject.projectId] = rseProject
+    
+    def printdebug(str, msgdebug):
```
Comment (Thurion) [inline RseData.py]: move into RseData class

File: RseData.py [hunk 2]
```diff
@@ -122,6 +122,8 @@ class RseData(object):
 
     VERSION = "1.3"
     VERSION_CHECK_URL = "https://api.github.com/repos/Thurion/EDSM-RSE-for-EDMC/releases"
+    
+    debug = False
```
Comment (Thurion) [inline RseData.py]: move into rseData class. I'm not sure if the value is actually stored.
It's also really out of place there :)

File: Backgroundworker.py [hunk 3]
```diff
@@ -56,8 +54,7 @@ def run(self):
             self.queue.task_done()
 
         if self.timer:
-            if __debug__:
-                print("EDSM-RSE: Stopping RSE background timer")
+            RseData.printdebug("EDSM-RSE: Stopping RSE background timer", True)
```
Comment (Thurion) [inline Backgroundworker.py]: remove EDSM-RSE

File: Backgroundworker.py [hunk 4]
```diff
@@ -34,8 +33,7 @@ def __init__(self, queue, rseData, interval=60 * 15):
         self.timer = None
 
     def timerTask(self):
-        if __debug__:
-            print("EDSM-RSE: timerTask triggered")
+        RseData.printdebug("EDSM-RSE: timerTask triggered", True)
```
Comment (Thurion) [inline Backgroundworker.py]: remove EDSM-RSE

File: BackgroundTask.py [hunk 5]
```diff
@@ -134,20 +132,20 @@ def queryEDSM(self, systems):
...
         if system:  # arrived in system without coordinates
-            if __debug__: print("EDSM-RSE: arrived in {}".format(system.name))
+            RseData.printdebug("EDSM-RSE: arrived in {}".format(system.name), True)
             system.removeFromProject(RseData.PROJECT_RSE)
             self.removeSystems()
 
         if not self.rseData.generateListsFromRemoteDatabase(*self.coordinates):
             # distances need to be recalculated because we couldn't get a new list from the database
-            if __debug__: print("EDSM-RSE: Using cached system list for distances. Radius was set to {}".format(self.rseData.calculateRadius()))
+            RseData.printdebug("EDSM-RSE: Using cached system list for distances. Radius was set to {}".format(self.rseData.calculateRadius()), True)
```
Comment (Thurion) [inline BackgroundTask.py]: remove EDSM-RSE

File: BackgroundTask.py [hunk 6]
```diff
@@ -134,20 +132,20 @@ def queryEDSM(self, systems):
...
-                if __debug__: print_exc()
+                RseData.printdebug("EDSM query call failed.", True)
         return set()
 
     def execute(self):
         system = self.getSystemFromID(self.systemAddress)
 
         if system:  # arrived in system without coordinates
-            if __debug__: print("EDSM-RSE: arrived in {}".format(system.name))
+            RseData.printdebug("EDSM-RSE: arrived in {}".format(system.name), True)
```
Comment (Thurion) [inline BackgroundTask.py]: remove EDSM-RSE

File: load.py [hunk 7]
```diff
@@ -111,7 +111,9 @@ def plugin_start(plugin_dir):
     this.clipboard = tk.BooleanVar(value=((settings >> 5) & 0x01))
     this.overwrite = tk.BooleanVar(value=((settings >> 6) & 0x01))
     this.edsmBodyCheck = tk.BooleanVar(value=not ((settings >> 7) & 0x01))  # invert to be on by default
-
+    this.debug = tk.BooleanVar(value=((settings >> 8) & 0x01))
+    RseData.setdebug(this.debug.get())
```
Comment (Thurion) [inline load.py]:
```suggestion
    this.rseData.debug(this.debug.get())
```

File: load.py [hunk 8]
```diff
@@ -257,6 +262,8 @@ def prefs_changed(cmdr, is_beta):
             this.queue.put(BackgroundTask.JumpedSystemTask(this.rseData, this.currentSystem))
 
     config.set(this.CONFIG_IGNORED_PROJECTS, this.rseData.ignoredProjectsFlags)
+    
+    RseData.setdebug(this.debug.get())
```
Comment (Thurion) [inline load.py]:
```suggestion
    this.rseData.debug(this.debug.get())
```

File: load.py [hunk 9]
```diff
@@ -238,7 +242,8 @@ def prefs_changed(cmdr, is_beta):
     # 6: copy to clipboard
     # 7: overwrite enabled status
     # 8: EDSM body check, value inverted
-    settings = (this.clipboard.get() << 5) | (this.overwrite.get() << 6) | ((not this.edsmBodyCheck.get()) << 7)
+    # 9: Debug
+    settings = (this.clipboard.get() << 5) | (this.overwrite.get() << 6) | ((not this.edsmBodyCheck.get()) << 7) | (this.debug.get() <<8)
```
Comment (Thurion) [inline load.py]:
```suggestion
    settings = (this.clipboard.get() << 5) | (this.overwrite.get() << 6) | ((not this.edsmBodyCheck.get()) << 7) | (this.debug.get() << 8)
```

File: RseData.py [hunk 10]
```diff
@@ -387,3 +388,15 @@ def initialize(self):
...
+    def printdebug(str, msgdebug):
+        if msgdebug == False:
+            print("EDSM-RSE: {0}".format(str))
+            return
+        
+        if msgdebug == True and RseData.debug == True:
+            print("EDSM-RSE(Debug): {0}".format(str))
+            return
+            
+    def setdebug(set):
```
Comment (Thurion) [inline RseData.py]: setdebug function not required when part of RseData class

File: Backgroundworker.py [hunk 11]
```diff
@@ -34,8 +33,7 @@ def __init__(self, queue, rseData, interval=60 * 15):
         self.timer = None
 
     def timerTask(self):
-        if __debug__:
-            print("EDSM-RSE: timerTask triggered")
+        RseData.printdebug("timerTask triggered", True)
```
Comment (Thurion) [inline Backgroundworker.py]:
```suggestion
        self.rseData.printdebug("timerTask triggered", True)

```

File: Backgroundworker.py [hunk 12]
```diff
@@ -56,8 +54,7 @@ def run(self):
             self.queue.task_done()
 
         if self.timer:
-            if __debug__:
-                print("EDSM-RSE: Stopping RSE background timer")
+            RseData.printdebug("Stopping RSE background timer", True)
```
Comment (Thurion) [inline Backgroundworker.py]:
```suggestion
            self.rseData.printdebug("Stopping RSE background timer", True)

```

File: RseData.py [hunk 13]
```diff
@@ -191,10 +193,9 @@ def openLocalDatabase(self):
             self.localDbConnection = sqlite3.connect(os.path.join(self.pluginDir, "cache.sqlite"))
             self.localDbCursor = self.localDbConnection.cursor()
         except Exception as e:
-            if __debug__:
-                print("EDSM-RSE: Local cache database could not be opened")
-            plug.show_error("EDSM-RSE: Local cache database could not be opened")
-            sys.stderr.write("EDSM-RSE: Local cache database could not be opened\n")
+            RseData.printdebug("Local cache database could not be opened", True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
            self.printdebug("Local cache database could not be opened", True)

```

File: RseData.py [hunk 14]
```diff
@@ -215,7 +216,7 @@ def adjustRadiusExponent(self, numberOfSystems):
             self.radiusExponent = int(self.radiusExponent) + 1
             if self.radiusExponent > RseData.MAX_RADIUS:
                 self.radiusExponent = 10
-            if __debug__: print("EDSM-RSE: Found too few systems, increasing radius to {1}".format(numberOfSystems, self.calculateRadius()))
+            RseData.printdebug("Found too few systems, increasing radius to {1}".format(numberOfSystems, self.calculateRadius()), True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
            self.printdebug("Found too few systems, increasing radius to {1}".format(numberOfSystems, self.calculateRadius()), True)

```

File: RseData.py [hunk 15]
```diff
@@ -224,7 +225,7 @@ def adjustRadiusExponent(self, numberOfSystems):
                 self.radiusExponent = 0
             if self.radiusExponent > RseData.MAX_RADIUS:  # prevent large radius after calculating on cached systems after switching a commander
                 self.radiusExponent = 10
-            if __debug__: print("EDSM-RSE: Found too many systems, decreasing radius to {1}".format(numberOfSystems, self.calculateRadius()))
+            RseData.printdebug("Found too many systems, decreasing radius to {1}".format(numberOfSystems, self.calculateRadius()), True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
            self.printdebug("Found too many systems, decreasing radius to {1}".format(numberOfSystems, self.calculateRadius()), True)

```

File: RseData.py [hunk 16]
```diff
@@ -275,12 +276,12 @@ def generateListsFromRemoteDatabase(self, cmdr_x, cmdr_y, cmdr_z):
             url = urlopen(rseUrl, timeout=10)
             if url.getcode() != 200:
                 # some error occurred
-                if __debug__: print("EDSM-RSE: error fetching nearby systems. HTTP code: " + url.getcode())
+                RseData.printdebug("error fetching nearby systems. HTTP code: " + url.getcode(), True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
                self.printdebug("error fetching nearby systems. HTTP code: " + url.getcode(), True)

```

File: RseData.py [hunk 17]
```diff
@@ -275,12 +276,12 @@ def generateListsFromRemoteDatabase(self, cmdr_x, cmdr_y, cmdr_z):
...
             if url.getcode() != 200:
                 # some error occurred
-                if __debug__: print("EDSM-RSE: error fetching nearby systems. HTTP code: " + url.getcode())
+                RseData.printdebug("error fetching nearby systems. HTTP code: " + url.getcode(), True)
                 return False
             response = url.read()
         except Exception as e:
             # some error occurred
-            if __debug__: print("EDSM-RSE: error fetching nearby systems: " + str(e))
+            RseData.printdebug("error fetching nearby systems: " + str(e), True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
            self.printdebug("error fetching nearby systems: " + str(e), True)

```

File: RseData.py [hunk 18]
```diff
@@ -317,7 +318,7 @@ def generateListsFromRemoteDatabase(self, cmdr_x, cmdr_y, cmdr_z):
         systems.sort(key=lambda l: l.distance)
 
         self.systemList = systems
-        if __debug__: print("EDSM-RSE: found {systems} systems within {radius} ly".format(systems=len(systems), radius=self.calculateRadius()))
+        RseData.printdebug("found {systems} systems within {radius} ly".format(systems=len(systems), radius=self.calculateRadius()), True)
```
Comment (Thurion) [inline RseData.py]:
```suggestion
        self.printdebug("found {systems} systems within {radius} ly".format(systems=len(systems), radius=self.calculateRadius()), True)

```

File: RseData.py (hunk 1 above)
Comment (Thurion) [inline RseData.py]:
```suggestion
    def printdebug(self, str, msgdebug):

```

File: BackgroundTask.py [hunk 19]
```diff
@@ -43,13 +43,11 @@ def __init__(self, rseData):
         self.rseData = rseData
 
     def execute(self):
-        if __debug__:
-            print("EDSM-RSE: {} didn't implement execute".format(self.__class__.__name__))
+        RseData.printdebug("{} didn't implement execute".format(self.__class__.__name__), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
        self.rseData.printdebug("{} didn't implement execute".format(self.__class__.__name__), True)

```

File: BackgroundTask.py [hunk 20]
```diff
@@ -43,13 +43,11 @@ def __init__(self, rseData):
...
     def execute(self):
-        if __debug__:
-            print("EDSM-RSE: {} didn't implement execute".format(self.__class__.__name__))
+        RseData.printdebug("{} didn't implement execute".format(self.__class__.__name__), True)
         pass  # to be implemented by subclass
 
     def fireEvent(self):
-        if __debug__:
-            print("EDSM-RSE: {} didn't implement fireEvent".format(self.__class__.__name__))
+        RseData.printdebug("{} didn't implement fireEvent".format(self.__class__.__name__), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
        self.rseData.printdebug("{} didn't implement fireEvent".format(self.__class__.__name__), True)

```

File: BackgroundTask.py [hunk 21]
```diff
@@ -75,7 +73,7 @@ def getSystemFromID(self, id64):
 
     def removeSystems(self):
         removeMe = list(filter(lambda x: len(x.getProjectIds()) == 0, self.rseData.systemList))
-        if __debug__: print("EDSM-RSE: adding {count} systems to removal filter: {systems}".format(count=len(removeMe), systems=[x.name for x in removeMe]))
+        RseData.printdebug("adding {count} systems to removal filter: {systems}".format(count=len(removeMe), systems=[x.name for x in removeMe]), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
        self.rseData.printdebug("adding {count} systems to removal filter: {systems}".format(count=len(removeMe), systems=[x.name for x in removeMe]), True)

```

File: BackgroundTask.py [hunk 22]
```diff
@@ -116,7 +114,7 @@ def queryEDSM(self, systems):
                 addToCache.append(system.id64)
         edsmUrl += "&".join(params)
 
-        if __debug__: print("EDSM-RSE: querying EDSM for {} systems".format(len(params)))
+        RseData.printdebug("querying EDSM for {} systems".format(len(params)), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
        self.rseData.printdebug("querying EDSM for {} systems".format(len(params)), True)

```

File: BackgroundTask.py [hunk 23]
```diff
@@ -134,20 +132,20 @@ def queryEDSM(self, systems):
                 return names
             except:
                 # ignore. the EDSM call is not required
-                if __debug__: print_exc()
+                RseData.printdebug("EDSM query call failed.", True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
                self.rseData.printdebug("EDSM query call failed.", True)

```

File: BackgroundTask.py [hunk 24]
```diff
@@ -134,20 +132,20 @@ def queryEDSM(self, systems):
...
-                if __debug__: print_exc()
+                RseData.printdebug("EDSM query call failed.", True)
         return set()
 
     def execute(self):
         system = self.getSystemFromID(self.systemAddress)
 
         if system:  # arrived in system without coordinates
-            if __debug__: print("EDSM-RSE: arrived in {}".format(system.name))
+            RseData.printdebug("arrived in {}".format(system.name), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
            self.rseData.printdebug("arrived in {}".format(system.name), True)

```

File: BackgroundTask.py [hunk 25]
```diff
@@ -134,20 +132,20 @@ def queryEDSM(self, systems):
...
             system.removeFromProject(RseData.PROJECT_RSE)
             self.removeSystems()
 
         if not self.rseData.generateListsFromRemoteDatabase(*self.coordinates):
             # distances need to be recalculated because we couldn't get a new list from the database
-            if __debug__: print("EDSM-RSE: Using cached system list for distances. Radius was set to {}".format(self.rseData.calculateRadius()))
+            RseData.printdebug("Using cached system list for distances. Radius was set to {}".format(self.rseData.calculateRadius()), True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
            self.rseData.printdebug("Using cached system list for distances. Radius was set to {}".format(self.rseData.calculateRadius()), True)

```

File: BackgroundTask.py [hunk 26]
```diff
@@ -273,14 +271,14 @@ def __init__(self, rseData, systemName, bodyCount, progress):
     def queryEdsm(self):
         edsmUrl = "https://www.edsm.net/api-system-v1/bodies?systemName={name}".format(name=quote(self.systemName))
         if __debug__:
-            print("EDSM-RSE: querying EDSM for bodies of system {}".format(self.systemName))
+            print("querying EDSM for bodies of system {}".format(self.systemName))
```
Comment (Thurion) [inline BackgroundTask.py]: missed call of debug output

File: BackgroundTask.py [hunk 27]
```diff
@@ -273,14 +271,14 @@ def __init__(self, rseData, systemName, bodyCount, progress):
...
-            print("EDSM-RSE: querying EDSM for bodies of system {}".format(self.systemName))
+            print("querying EDSM for bodies of system {}".format(self.systemName))
         try:
             url = urlopen(edsmUrl, timeout=10)
             response = url.read()
             edsmJson = json.loads(response)
             return edsmJson["id64"], len(edsmJson["bodies"])
         except:
-            if __debug__: print_exc()
+            RseData.printdebug("EDSM body count call failed.", True)
```
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
            self.rseData.printdebug("EDSM body count call failed.", True)

```

File: BackgroundTask.py (hunk 26 above)
Comment (Thurion) [inline BackgroundTask.py]:
```suggestion
        self.rseData.printdebug("querying EDSM for bodies of system {}".format(self.systemName))

```

**Review by Thurion — CHANGES_REQUESTED** [review]

File: BackgroundTask.py (hunk 6 above)
Comment (RapidfireCRH) [reply BackgroundTask.py]: This change was completed

File: BackgroundTask.py (hunk 5 above)
Comment (RapidfireCRH) [reply BackgroundTask.py]: This change was completed

File: Backgroundworker.py (hunk 4 above)
Comment (RapidfireCRH) [reply Backgroundworker.py]: This change was completed

File: Backgroundworker.py (hunk 3 above)
Comment (RapidfireCRH) [reply Backgroundworker.py]: This change was completed

**Review by Thurion — APPROVED** [review]

**Pull request merged**